lib/ddtrace/span.rb

```diff
@@ -91,6 +92,9 @@ def set_error(e)
 
     # Mark the span finished at the current time and submit it.
     def finish
+      return if @finished
+      @finished = true
+
```

Comment:
Generally speaking, I'm all in for this, finish should be done only once. Then I'm a little worried about what it means to call `finish_at`. Because, here, `finish_at` can still be called many times, so its name becomes quite misleading. I don't know if it makes sense to move this test to `finish_at` so that this function too, can not be call twice. And the other question is, do we need the boolean or would `!@end_time.nil?` be enough to consider is "finishable".